src/modules/utf8.js

```diff
@@ -63,8 +67,16 @@ import { toPDFName } from "../libs/pdfname.js";
         map[code] !== null &&
         typeof map[code].toString === "function"
       ) {
-        unicode = ("0000" + map[code].toString(16)).slice(-4);
-        code = ("0000" + (+code).toString(16)).slice(-4);
+        unicode = map[code];
+        if (unicode > 0xffff) {
+          unicode -= 0x10000;
+          unicode =
+            ((unicode >> 10) + 0xd800).toString(16).padStart(4, "0") +
+            ((unicode % 0x400) + 0xdc00).toString(16).padStart(4, "0");
+        } else {
+          unicode = unicode.toString(16).padStart(4, "0");
+        }
+        code = (+code).toString(16).padStart(4, "0");
```

Comment:
This is an implementation for outputting surrogate pairs

I see [PDF 1.3 specification](https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/pdfreference1.3.pdf) because README says to follow PDF 1.3 specification.

It says that in 3.8.1

> The remainder of the string consists of Unicode character codes, according to the UTF-16 encoding
specified in the Unicode standard, version 2.0.

So I output code as UTF-16 surrogate pairs encoding.